gslab_scons/tests/test_misc.py

```diff
@@ -1,95 +1,305 @@
-    	self.assertEqual(misc.is_unix(), not sys.platform.startswith("win"))
+        '''
+        Test that is_unix() returns True on Mac and Linux
+        machines and False otherwise. 
+        '''
+        platform_ref = 'gslab_scons.misc.sys.platform'
+        with mock.patch(platform_ref, 'win32'):
+            self.assertFalse(misc.is_unix())
+
+        with mock.patch(platform_ref, 'darwin'):
+            self.assertTrue(misc.is_unix())  
+
+        with mock.patch(platform_ref, 'linux'):
+            self.assertTrue(misc.is_unix())    
+
+        with mock.patch(platform_ref, 'atheos'):
+            self.assertFalse(misc.is_unix())     
 
     def test_is_64_windows(self):
-    	pass
+        '''
+        Test that is_64_windows() returns True when PROGRAMFILES(X86)
+        is a key in the system environment as accessed through 
+        os.environ.
+        '''
+        environ = 'gslab_scons.misc.os.environ'
+        mocked_environ = {'PROGRAMFILES(X86)': 'C:/Program Files (x86)'}
+        with mock.patch(environ, mocked_environ):
+            self.assertTrue(misc.is_64_windows())
+        with mock.patch(environ, dict()):
+            self.assertFalse(misc.is_64_windows())
 
-    def test_command_line_arg(self):
-        self.assertEqual(misc.command_line_arg({'test' : ''}), '')
-        self.assertEqual(misc.command_line_arg({'CL_ARG' : 'Test'}), 'Test')
+    @mock.patch('gslab_scons.misc.os.environ', {'PATH': '/bin:usrs/local'})
+    @mock.patch('gslab_scons.misc.os.pathsep', ':')
+    @mock.patch('gslab_scons.misc.os.access')
+    def test_is_in_path(self, mock_access):
+        '''
+        Test that is_in_path() returns i) the full path of 
+        a recognised executable that the function takes as
+        an argument and ii) None if the function's argument
+        is not recognised as an executable. 
+        '''
+        mock_access.side_effect = self.access_side_effect
+
+        self.assertEqual(misc.is_in_path('stata'), 
+                         '/bin/stata')
+        self.assertEqual(misc.is_in_path('executable_file'), 
+                         'executable_file')
+        self.assertEqual(misc.is_in_path('stata-mp'),
+                         None)
+                            
+    @staticmethod
+    def access_side_effect(*args, **kwargs):
+        '''
+        This function mocks os.access by defining which
+        files our mocked up machine can execute or 
+        otherwise access. 
+        '''
+        # Define the files to which we have access
+        # Total access        
+        execute_files = ['/bin/stata',
+                         'executable_file']
+        # Total access except execution
+        other_files   = ['test_script.do']
+        
+        # Access the arguments of the os.access() call.
+        path = args[0]
+        mode = args[1]
+
+        if mode == os.X_OK and path in execute_files:
+            result = True
+        elif mode != os.X_OK and path in (execute_files + other_files):
```

Comment:
`os.access()` is only ever called with `mode = os.X_OK` in `misc.py`, but I thought I'd make the mock more realistic by allowing different modes so that it would catch changes we may make (i.e. changing the the `mode` argument from `os.access()` in `misc.py`).

I am happy to get rid of this if you think it adds unnecessary complexity!